pkg/claims/migrateClaimsWrapper.go

```diff
@@ -35,33 +54,184 @@ func (w *migrateClaimsWrapper) Store() crud.Store {
 	return w.storeWithConnect
 }
 
-func (w *migrateClaimsWrapper) Read(itemType string, name string) ([]byte, error) {
-	// Read is also called during List (ReadAll) so this will migrate everything for a list too
-	return w.Migrate(itemType, name)
+func (w *migrateClaimsWrapper) List(itemType string, group string) ([]string, error) {
+	names, err := w.Store().List(itemType, group)
+	if err != nil {
+		if strings.Contains(err.Error(), crud.ErrRecordDoesNotExist.Error()) {
+
+			// Proactively migrate all data in the claims/ dir if we detect the layout is old
+			err := w.MigrateAll()
+			if err != nil {
+				return nil, err
+			}
+
+			return w.Store().List(itemType, group)
+
+		}
+
+		return nil, err
+	}
+
+	return names, nil
 }
 
-func (w *migrateClaimsWrapper) Migrate(itemType string, name string) ([]byte, error) {
+func (w *migrateClaimsWrapper) Read(itemType string, name string) ([]byte, error) {
 	data, err := w.Store().Read(itemType, name)
 	if err != nil {
+		// If we can't read the data, check for the data layout migration from rewriting the claims spec
+		if getSchemaVersion(data) != claim.CNABSpecVersion {
+
+			// Proactively migrate all data in the claims/ dir if we detect the layout is old
+			err := w.MigrateAll()
+			if err != nil {
+				return nil, err
+			}
+
+			return w.Store().Read(itemType, name)
+		}
+
 		return nil, err
 	}
 
-	//  We only migrate claim data, ignore the rest
-	if itemType != claim.ItemTypeClaims {
-		return data, nil
+	return data, nil
+}
+
+func (w *migrateClaimsWrapper) MigrateAll() error {
+	fmt.Fprint(w.Err, "!!! Migrating claims data to match the CNAB Claim spec https://cdn.cnab.io/schema/cnab-claim-1.0.0-DRAFT+b5ed2f3/claim.schema.json. This is a one-way migration !!!\n")
```

Comment:
Should we Fprintf using `claim.CNABSpecVersion` instead of the hard-coded version here?

Reply:
This migration is specific to getting them from the before times to cnab-claim-1.0.0-DRAFT+d7ffba8. When we bump claims down the road, I think we may have a different migration. I am not sure we want to automatically tie this together just yet since it's not a generic migration.